app/services/foreman/plugin.rb

```diff
@@ -93,6 +93,17 @@ def installed?(id)
 
     def initialize(id)
       @id = id.to_sym
+      add_loggers
+    end
+
+    def add_loggers
+      return unless (defined?(Foreman::Logging) && SETTINGS[@id])
+
+      if (loggers = SETTINGS[@id][:loggers])
```

Comment:
An example of this we use today is for Pulp rest call logging. We hand the Runcible library that makes our calls down to Pulp a logger and then can disable/enable it (or if undefined, nothing would happen) - https://github.com/Katello/katello/blob/master/app/lib/katello/util/thread_session.rb#L67

One question I am not sure on is whether plugins should be able to define per environment settings for their loggers. Per environment settings was something I missed in the first pass of this implementation (and resulted in the massive log files and console printing) and will be accounted for in the next push.